rcv/frame_handlers/frc_2019.py

```diff
@@ -0,0 +1,240 @@
+import cv2
+import math
+import numpy as np
+import os
+import sys
+
+canstants_path = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+sys.path.append(canstants_path)
+from constants import X_PIXELS_PER_DEGREE, Y_PIXELS_PER_DEGREE
+
+response_structure = {
+        "front": False,
+        "found": False,
+        "p1": {"d": 0, "a": 0},
+        "p2": {"d": 0, "a": 0},
+    }
+
+
+class FrameHandler():
+    def __init__(self, configuration):
+        self._conf = configuration
+        self._frame = None
+        self._cleaned_frame = None
+        self._line_contour = None
+        self._points = None
+        self._line_contour = None
+        self._y_leaning_angle = None
+        self._x_turning_angle = None
+        self._response = None
+        self._show = bool(self._conf.get('rcv_server', 'show'))
+        focus_image_path = self._conf.get('rcv_server', 'focus_image')
+        if os.path.exists(focus_image_path) and os.path.isfile(focus_image_path):
+            # TODO do we really need the colored focus image
+            self._focus_image = cv2.imread(focus_image_path)
+            self._focus_image_gray = cv2.imread(focus_image_path, cv2.IMREAD_GRAYSCALE)
+        else:
+            raise FileNotFoundError("File {} was not found")
+        print("#### Initialized FrameHandler .......")
+
+    def init_frame_analysis(self):
+        self._frame = None
+        self._cleaned_frame = None
+        self._line_contour = None
+        self._y_leaning_angle = None
+        self._x_turning_angle = None
+        self._response = response_structure.copy()
+
+    def line_distance(self, p1, p2):
+        x1 = p1[0]
+        x2 = p2[0]
+        y1 = p1[1]
+        y2 = p2[1]
+        return math.sqrt((x1-x2)**2 + (y1 - y2)**2)
+
+    def averge_point(self, p1, p2):
+        x1 = p1[0]
+        x2 = p2[0]
+        y1 = p1[1]
+        y2 = p2[1]
+        return(int((x1 + x2)/2), int((y1 + y2)/2))
+
+    def show_images(self):
```

Comment:
I think that maybe this function should be called progress_images and it would return a list of tuples that are images, and the names of the windows they would be displayed in. The nes56CvManager would call it to get them and then it will show them and then FrameHandler wouldn't need to know whether it should show itself or not

Reply:
Not sure what you mean by that ...

Reply:
the function itself wouldn't show the images, it would return a list of the images that it wants to show to the user and then nes56_cv_manager would show them so frc_2019 wouldn't need to know if it shows itself